flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/Table.java

```diff
@@ -472,14 +564,31 @@
 	 *   table.joinLateral("split(c) as (s)").select("a, b, c, s");
 	 * }
 	 * </pre>
+	 * @deprecated use {@link #joinLateral(Expression)}
 	 */
+	@Deprecated
 	Table joinLateral(String tableFunctionCall);
 
 	/**
 	 * Joins this {@link Table} with an user-defined {@link TableFunction}. This join is similar to
 	 * a SQL inner join with ON TRUE predicate but works with a table function. Each row of the
 	 * table is joined with all rows produced by the table function.
 	 *
+	 * <p>Example:
+	 *
+	 * <pre>
+	 * {@code
+	 *   class MySplitUDTF extends TableFunction<String> {
+	 *     public void eval(String str) {
+	 *       str.split("#").forEach(this::collect);
+	 *     }
+	 *   }
+	 *
+	 *   TableFunction<String> split = new MySplitUDTF()
+	 *   table.joinLateral(call(split, $("c")).as("s")).select($("a"), $("b"), $("c"), $("s"))
```

Comment:
use `MySplitUDTF.class`? we should advocate the shorter approach without an instance

Reply:
We do not have the `call(Class<? extends UserDefinedFunction> function, Object... arguments)`. I must have missed that. I will adress that in a follow up if you are fine with that.

Reply:
Follow up issue: https://issues.apache.org/jira/browse/FLINK-17523